### ISSUE TITLE
TOML Support

### ISSUE THREAD

**Issue by helloanoop** [usebruno/bruno#1303]
### Overview
This discussion aims to explore the possibility of using `.toml` as the preferred format for saving API request data instead of `.bru`. For the progression of the Domain-Specific Language (DSL), see https://github.com/usebruno/bruno/discussions/360

I've been seriously exploring how we can represent api requests in `toml`
After careful consideration, factoring in various edge cases, `toml` seems to fulfil our requirements and might offer better long-term benefits.

### Why TOML
- Well established format
- Utilize syntax highlighting in IDEs
- Leverage existing tooling for toml file manipulation
- Git providers (like GitHub, Bitbucket) offer syntax highlighting while reviewing pull requests on the web

### Tasks
 - [ ] Experimental support for working with TOML in GUI
 - [ ] Experimental support for working with TOML in CLI
 - [ ] Gather Community feedback during our community call on Wed Jan 3rd 2024
 
More Tasks TBD based on how the experiment turns out

### Looming Issues
Several features are currently blocked due to limitations in the bru language. Some include:

1. Describing headers
2. Managing choices for query parameters
3. File upload capabilities
4. Storing multiple payloads within a single request

To address these limitations, I see two potential paths:
1. Enhance Bru Lang
2. Utilize TOML

### Next Steps
Before choosing a direction, I aim to implement experimental TOML support in our current app within the next 3-4 days (by January 1st, 2024). . BTW I am working on bruno [Full Time](https://www.usebruno.com/blog/bootstrapping) from Jan 1st :)

### Sample GET Request
![image](https://github.com/usebruno/bruno/assets/9350733/fa0f8a82-550f-460c-b539-c0816b0118c1)

**Comment by kishaningithub** [comment]
I feel that TOML should be the way forward.  Would appreciate more articles on TOML vs bru as you move forward with the next steps.

**Comment by nateshmbhat** [comment]
any update on this ? this seems very interseting

**Comment by hp77-creator** [comment]
Hey @helloanoop, just wanted to know did you also explore/consider YAML for the language?

**Comment by hp77-creator** [comment]
> Hey @helloanoop, just wanted to know did you also explore/consider YAML for the language?

My bad, went through #360 , indentation hell 🤔  interesting, I mean at one point it can become that but then at that point your API design is not good, It will be on the type of API imo.

**Comment by jzorn** [comment]
I think TOML is a better format for humans. It's too easy to mess things up in YAML.

**Comment by DenisMir** [comment]
Any progress on this? There are so much bugs just going back to the custom bru DSL. It would be awesome to get rid of all of them. (e.g. the colon bug)

**Comment by helloanoop** [comment]
We will move very soon (max 2 weeks) to support YAML to save the requests

@lohxt1 on our team is already working on it.
There is a lot of nuance on TOML support, I will add detailed notes later

TLDR; - YAML support to arrive soon, TOML will need more time

**Comment by aarjithn** [comment]
@helloanoop my 2c is to stick with just one. If its yaml so be it. Who are very particular can setup conversion to/from toml

Maybe depreciate and support bru for a while but going down path of multiple config languages would be a pain

**Comment by skhaz** [comment]
I do not think a custom language for settings a good idea.

Please try to stick to toml or yaml, do not reinvent the wheel.

**Comment by DenisMir** [comment]
Yeah. E.g. not allowing to have "colons" in your custom format for values is a bad joke. That is why inventing some custom configuration format is definitely a pain and just leads to bugs that normally would not be a problem with a basic standard format.

**Comment by mabramo** [comment]
I agree one standard should be developed for. Bru is nice, with some issues. Personally I much prefer TOML over YAML as I find it easier to read.

**Comment by trymeouteh** [comment]
If TOML can work for all Bruno usecases. I am for it. This way we can choose which formatter tool to format the TOML files.

**Comment by helloanoop** [comment]
We ultimately decided to adopt YAML over TOML

TOML works well for small configurations, but becomes hard to manage with deeply nested structures and arrays. As a result, we chose YAML.

Our collection schema is pretty large enough that we built a [visual schema explorer](https://schema.opencollection.com/) in-house to make the format easier to understand and adopt. TOML becomes unreadable at this level.

Also, Bruno v3 now supports YAML 🎉 🎉 
Download the latest at https://www.usebruno.com/downloads
Join the discussion at https://github.com/usebruno/bruno/discussions/6634

**Comment by kishaningithub** [comment]
@helloanoop How to make bruno switch existing files from .bru to YAML?

**Comment by helloanoop** [comment]
@kishaningithub For now
1. Click on share from collection context menu
2. Choose OpenCollection export as format
3. ReImport in Bruno. Make sure to choose file format as yml
